pkgs/development/tools/parsing/tree-sitter/default.nix

```diff
@@ -47,6 +47,17 @@ let
 
   buildGrammar = callPackage ./grammar.nix { };
 
+  grammarsSrc' = import ./grammars { inherit lib; } // extraGrammars;
```

Comment:
Why these lines were introduced?
They look like adding a functionality not directly related to the title of this pull request. If this is the case, split them into a separate commit, and succinctly explain why they are needed.